File: Solutions/2319/2319.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 2319.c
 * Author: washing
 * Version: 1.0
 * Date: 2023-01-31 10:26:53 
***************************************************************************************************/


bool checkXMatrix(int** grid, int gridSize, int* gridColSize){
    for (int i=0; i<gridSize; i++) {
        for (int j=0; j<gridSize; j++) {
            if (j == i | j == gridSize-1-i) {
                if (grid[i][j] == 0) return false;
            }
            else if (grid[i][j] != 0) {
                return false;
            }
        }
    }
    return true;
}

File: Solutions/2319/2319.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2023/1/31 10:21
# File: 2319.py
# Desc: 


class Solution:
    def checkXMatrix(self, grid: List[List[int]]) -> bool:
        n = len(grid)
        for i in range(n):
            if grid[i][i] == 0 or grid[i][n-1-i] == 0:
                return False
            if i == n-1-i:
                if grid[i][i] != sum(grid[i]):
                    return False
            elif grid[i][i]+grid[i][n-1-i] != sum(grid[i]):
                return False
        return True
